### ISSUE TITLE
Local function completion is broken in certain top-level statements

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#67613]
**Version Used**:  VS 17.6 Preview 2

Expected `function` to appear in the completion list in both the first position (preceding local function decl) and the second position (preceding call to Console.WriteLine).

In practice, `function` only appears in the second position.

![top-level-local-function-lookup](https://user-images.githubusercontent.com/5833655/229382183-58a03b9f-37d3-4915-b20f-90bcbff33fdd.gif)

**Comment by CyrusNajmabadi** [comment]
I don't believe ide does anything special here. We'll just show what the compiler returns in LookupSymbols

**Comment by jcouv** [comment]
@RikkiGibson I suspect this is an error recovery problem. When typing right before the local function, we're not able to parse the local function properly so it doesn't appear as an available symbol. One mitigation is to add a return statement just before the local function. That helps with parser recovery.
I'm going to move this to backlog as I don't have a good idea how we could fix this. Maybe we could use line breaks as hints for parser recovery? Feel free to move this sooner if you'd like.

**Comment by CyrusNajmabadi** [comment]
yeah, it's def an error recovery scenario.  can certainly make better in the case of `id keyword id`, esp. when keyword is on the next line.  It will be harder to support things like:

```c#
func

T func ...
```

though.

**Comment by CyrusNajmabadi** [comment]
Note: does this work ok *inside* a method body?  If so, you should see what happens there and port that code to the top-level parsing code.
